### PR TITLE
[hail] add hl.shuffle

### PR DESCRIPTION
**Pull request by cseed**
CHANGELOG: added shuffle function

### CODE REVIEW

File: hail/python/hail/expr/functions.py [hunk 1]
```diff
@@ -5711,3 +5711,28 @@ def _values_similar(left, right, tolerance=1e-6, absolute=False):
...
+    a : :class:`.ArrayExpression`
+        Array to permute.
+    seed : :obj:`int`, optional
+        Random seed.
+
+    Returns
+    -------
+    :class:`.ArrayExpression`
+    """
+    return sorted(a, key=lambda _: hl.rand_unif(0.0, 1.0, seed=seed))
```
Comment (konradjk) [inline hail/python/hail/expr/functions.py:5738]: `hl.sorted`

Comment (tpoterba) [reply hail/python/hail/expr/functions.py:5738]: it's defined in this file

Comment (tpoterba) [reply hail/python/hail/expr/functions.py:5738]: you'd need to use `builtins.sorted` to get python base `sorted`.

Comment (konradjk) [reply hail/python/hail/expr/functions.py:5738]: oh i see. isn't it kinda bad form to override the builtins? i guess you're doing that a lot anyway...

Comment (tpoterba) [reply hail/python/hail/expr/functions.py:5738]: we only override the builtins in our code, not user code (unless people do `from hail import *`, which is definitely bad form). I think the predictability is a big win: `len` -> `hl.len`, `sorted` -> `hl.sorted`, etc

Comment (tpoterba) [reply hail/python/hail/expr/functions.py:5738]: (except for all the things we can't override, like and/or/if, of course)

Comment (konradjk) [reply hail/python/hail/expr/functions.py:5738]: yeah agreed it's a big win. i just wasn't sure if there was a better way to do this (like call it _sorted, but then be able to put it as hl.sorted), but i guess not

**Comment by cseed** [comment]
Thanks, fixed!

**Pull request merged**